core/message-handling.go

```diff
@@ -40,15 +40,19 @@ type messageStreamHandler func(in <-chan []byte, reply chan<- []byte)
 // message hasn't been processed before.
 type messageHandler func(msg interface{}) (reply <-chan interface{}, new bool, err error)
 
+// messageConsumer receives a generated message.
```

Comment:
I agree that `messageConsumer` might not be a good name for the thing. I would suggest naming it `generatedMessageConsumer` or `generatedMessageHandler` with a one-line summary like "takes care of generated message" (but a bit more precise). Maybe later we decide to move message signing and UI assigning in there. Then "recorder" might not be the best word to reflect the purpose. 

I would also suggest we rename `messageHandler`rather to `incomingMessageHandler`. This would be a bit more consistent with the naming in `client` package.

Reply:
Sounds good